exec: panic on EXPERIMENTAL SCRUB

```
root@127.0.0.1:53303/defaultdb> SET experimental_vectorize=on;
SET

Time: 303µs

root@127.0.0.1:53303/defaultdb> CREATE TABLE t (
  id int PRIMARY KEY,
  name STRING,
  CONSTRAINT abc CHECK (name > 'he')
);
CREATE TABLE

Time: 2.781ms

root@127.0.0.1:53303/defaultdb> INSERT INTO t VALUES (1, 'hello');
INSERT 1

Time: 3.598ms

root@127.0.0.1:53303/defaultdb> EXPERIMENTAL SCRUB TABLE t;
E190702 20:29:29.942748 709 sql/conn_executor.go:713  [n1,client=127.0.0.1:53305,user=root] a SQL panic has occurred while executing "EXPERIMENTAL SCRUB TABLE t": interface conversion: coldata.column is []int64, not [][]uint8
E190702 20:29:29.942800 709 util/log/crash_reporting.go:207  [n1,client=127.0.0.1:53305,user=root] a panic has occurred!
panic: interface conversion: coldata.column is []int64, not [][]uint8 [recovered]
	panic: panic while executing 1 statements: EXPERIMENTAL SCRUB TABLE _; caused by interface conversion: coldata.column is []int64, not [][]uint8

goroutine 709 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc001247100, 0x75f07e0, 0xc000f6c640, 0x69eb2c0, 0xc001249620)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:727 +0x330
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc001247100, 0x75f07e0, 0xc000f6c640)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:440 +0x61
panic(0x69eb2c0, 0xc001249620)
	/usr/local/Cellar/go/1.12.6/libexec/src/runtime/panic.go:522 +0x1b5
github.com/cockroachdb/cockroach/pkg/sql/exec/coldata.(*memColumn).Bytes(0xc000a1c3c0, 0x0, 0x7652460, 0xc000a1c3c0)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/exec/coldata/vec.go:221 +0x74
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*materializer).Next(0xc0000c0a00, 0x592537c, 0xc000a1c440, 0xc001a8cdc0, 0x59b556e)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/materializer.go:170 +0xe47
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.Run(0x75f08a0, 0xc0012492f0, 0x75fc220, 0xc0000c0a00, 0x75cd520, 0xc001330000)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/base.go:171 +0x35
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*ProcessorBase).Run(0xc0000c0a00, 0x75f08a0, 0xc0012492f0)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/processors.go:793 +0x92
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*Flow).Run(0xc00164a780, 0x75f08a0, 0xc0012492f0, 0x6f49358, 0x0, 0x0)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow.go:654 +0x1f2
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc00063ad80, 0xc0016f3680, 0xc001410b40, 0xc001a8dba0, 0xc001330000, 0xc001409400, 0x0)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:296 +0x35c
github.com/cockroachdb/cockroach/pkg/sql.scrubRunDistSQL(0x75f08a0, 0xc0012481e0, 0xc0016f3680, 0xc001247450, 0xc001a8dba0, 0x96c9520, 0x3, 0x3, 0x0, 0x0, ...)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/scrub.go:564 +0x2f8
github.com/cockroachdb/cockroach/pkg/sql.(*physicalCheckOperation).Start(0xc0016f3320, 0x75f08a0, 0xc0012481e0, 0xc001408a00, 0xc001247450, 0x0, 0x0)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/scrub_physical.go:145 +0xa74
github.com/cockroachdb/cockroach/pkg/sql.(*scrubNode).Next(0xc000f665f0, 0x75f08a0, 0xc0012481e0, 0xc001408a00, 0xc001247450, 0x0, 0x75f08a0, 0xc0012481e0)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/scrub.go:127 +0x228
github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Next(0xc001406000, 0xc001247450, 0x75f2760, 0xc000f665f0, 0x0)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_node_to_row_source.go:171 +0x43c
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.Run(0x75f08a0, 0xc0012481e0, 0x75fbe20, 0xc001406000, 0x75cf0a0, 0xc000c5e000)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/base.go:171 +0x35
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*ProcessorBase).Run(0xc001406000, 0x75f08a0, 0xc0012481e0)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/processors.go:793 +0x92
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*Flow).Run(0xc00164a000, 0x75f08a0, 0xc0012481e0, 0x6f49358, 0x0, 0x0)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow.go:654 +0x1f2
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc00063ad80, 0xc0016f3200, 0xc001410b40, 0xc001a8e8b0, 0xc00154c2c0, 0xc001247528, 0x0)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:296 +0x35c
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun(0xc00063ad80, 0x75f08a0, 0xc001248060, 0xc001247528, 0xc0016f3200, 0xc001410b40, 0x75f2760, 0xc000f665f0, 0xc00154c2c0)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:860 +0x208
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0xc001247100, 0x75f08a0, 0xc001248060, 0xc001247450, 0x3, 0xc954f68, 0xc00052bb80, 0xc001a8eb00, 0x0, 0x0, ...)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:902 +0x370
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc001247100, 0x75f08a0, 0xc001248060, 0xc001247450, 0xc954f68, 0xc00052bb80, 0x0, 0x0)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:734 +0x6e5
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc001247100, 0x75f08a0, 0xc001248060, 0x75f5c60, 0xc001076780, 0xc000fad337, 0x1a, 0x0, 0x1, 0x0, ...)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:417 +0xb64
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc001247100, 0x75f08a0, 0xc001248060, 0x75f5c60, 0xc001076780, 0xc000fad337, 0x1a, 0x0, 0x1, 0x0, ...)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:99 +0x4ec
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc001247100, 0x75f07e0, 0xc000951440, 0x0, 0x0)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1204 +0x1ab0
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc001247100, 0x75f07e0, 0xc000f6c640, 0xc0006190c0, 0x5400, 0x15000, 0xc000619158, 0xc0004c2670, 0x0, 0x0)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1140 +0x1a3
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc0001e3680, 0x75f07e0, 0xc000f6c640, 0xc001247100, 0x5400, 0x15000, 0xc000619158, 0xc0004c2670, 0x0, 0x0)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:442 +0xce
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1(0xc0003e99c9, 0xc0010600a0, 0x75f07e0, 0xc000f6c640, 0xc0004c2670, 0xc0001e3680, 0xc00105c000, 0x75f4220, 0xc001060080, 0xc000f45e60, ...)
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:580 +0x21c
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	/Users/solon/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:509 +0x17b
```

--- a/pkg/sql/distsqlrun/column_exec_setup.go
+++ b/pkg/sql/distsqlrun/column_exec_setup.go
@@ -119,6 +119,9 @@ func newColOperator(
 		if err := checkNumIn(inputs, 0); err != nil {
 			return nil, nil, err
 		}
+		if core.TableReader.IsCheck {
+			return nil, nil, errors.Newf("scrub table reader is unsupported in vectorized")
+		}
 		op, err = newColBatchScan(flowCtx, core.TableReader, post)
 		// We want to check for cancellation once per input batch, and wrapping
 		// only colBatchScan with an exec.CancelChecker allows us to do just that.

--- a/pkg/sql/logictest/testdata/logic_test/vectorize
+++ b/pkg/sql/logictest/testdata/logic_test/vectorize
@@ -412,3 +412,13 @@ query I
 SELECT b FROM a WHERE b < 0.5
 ----
 0
+
+# Test unsupported scrub (should fall back to distsql).
+statement ok
+CREATE TABLE t38626 (id int PRIMARY KEY, name STRING, CONSTRAINT abc CHECK (name > 'he'))
+
+statement ok
+INSERT INTO t38626 VALUES (1, 'hello')
+
+statement ok
+EXPERIMENTAL SCRUB TABLE t38626

